cmd/common/registry.go

```diff
@@ -172,90 +177,96 @@ func buildCommandFromGadget(gadget gadgets.Gadget,
 
 			log := logger.DefaultLogger()
 
-			// Create new runner
-			runner := gadgetrunner.NewGadgetRunner(
+			gadgetContext := gadgetcontext.New(
 				ctx,
 				"",
 				runtime,
 				gadget,
+				gadgetParams,
 				parser,
 				log,
 			)
 
-			if parser != nil {
-				// Add some custom params like filters that are available when using parser
-				if len(filters) > 0 {
-					err = parser.SetFilters(filters)
-					if err != nil {
-						return err // TODO: Wrap
-					}
-					gadgetParams.AddKeyValuePair("columns_filters", strings.Join(filters, ",")) // TODO: maybe encode?! difficult for CRs though
+			if parser == nil {
+				// This kind of gadgets return directly the result instead of
+				// using the parser
+				result, err := runtime.RunGadget(gadgetContext, operatorsPerGadgetParamCollection)
```

Comment:
Any reason not to include `operatorsPerGadgetParamCollection` in the context too?